src/app/loadout-analyzer/hooks.tsx

```diff
@@ -41,18 +45,33 @@ const autoOptimizationContextSelector = currySelector(
     createItemContextSelector,
     unlockedPlugSetItemsSelector.selector,
     savedLoStatConstraintsByClassSelector,
-    allItemsSelector,
     autoModSelector,
-    (itemCreationContext, unlockedPlugs, savedLoStatConstraintsByClass, allItems, autoModDefs) =>
-      itemCreationContext.defs &&
-      autoModDefs &&
-      ({
-        itemCreationContext,
-        unlockedPlugs,
-        savedLoStatConstraintsByClass,
-        allItems,
-        autoModDefs,
-      } satisfies LoadoutAnalysisContext),
+    allItemsSelector,
+    loVendorItemsSelector.selector,
+    settingSelector<'loIncludeVendorItems'>('loIncludeVendorItems'),
```

Comment:
This doesn't type check without manually specifying the type parameter, which is weird, but probably not within our control.